lib/KTable/index.vue

```diff
@@ -283,6 +303,23 @@
         return colIndex => this.sortable && this.headers[colIndex].dataType !== DATA_TYPE_OTHERS;
       },
     },
+    watch: {
```

Comment:
First, thanks for covering this!

I wonder if it'd be just fine to have it in `mounted` or similar and be run once. Since `columnId`, as far as we know, won't be dynamically obtained but defined by devs right in the code. That said, I am not sure if the performance  cost of the watcher is so significant. Any thoughts?

Reply:
I see! I guess it's a matter of taste. I felt that lifecycle hooks would not be a good place to put the validation logic, as the hooks are called only once and thus if the prop values changes in future, then we will not get the same validations again. But I do see that the cases in which people are using code itself to generate the column ids would be rare. 

As for the performance cost, I don't think that they would be any overhead practically (as if the props aren't changing, the watcher would never run again, and if they, so it will be run every time they change, which should be the expected behaviour!)

Reply:
Thanks @EshaanAgg. I think we can leave this way then.